### PR TITLE
FIX: remove RBACs from target-namespace when ArgoCD instance is deleted

### PR DESCRIPTION
**Pull request by shubhamagarwal19**
Cherry-picked the fix added to rhos-v1.2 branch -> https://github.com/argoproj-labs/argocd-operator/pull/404

### CODE REVIEW

File: controllers/argocd/util_test.go [hunk 1]
```diff
@@ -460,3 +461,54 @@ func TestDeleteRBACsForNamespace(t *testing.T) {
...
+	r := makeTestReconciler(t)
+
+	ns := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{
+		Name: "testNamespace",
+		Labels: map[string]string{
+			common.ArgoCDManagedByLabel: a.Namespace,
+		}},
+	}
+
+	err := r.client.Create(context.TODO(), ns)
```
Comment (jannfis) [inline controllers/argocd/util_test.go:476]: Client method became public with Operator SDK upgrade, please check.

**Review by jannfis — COMMENTED** [review]
Thanks for the "back"-port to master, @shubhamagarwal19.

Just one thing, because the build & tests due to a unit test, please see below.

File: controllers/argocd/util_test.go (hunk 1 above)
Comment (shubhamagarwal19) [reply controllers/argocd/util_test.go:476]: ahhh thanks .. missed it for the test files!

**Review by iam-veeramalla — APPROVED** [review]
LGTM, Thanks @shubhamagarwal19  !!

**Pull request merged**